### ISSUE TITLE
Use tempfile by default, or have user specify outfile

### ISSUE THREAD

**Issue by dgrtwo** [jennybc/reprex#4]
Currently the function works in the current directory using a base filename along the lines of 08-25-2015_REPREX_gh. It keeps them there if outfiles = TRUE.

I suspect that:

* Most users won't want to keep the input and output files, especially since the goal is to get it into GitHub or SO
* Those that do would be at least as happy providing their own filename (so they can refer to it later); while they can specify 

Thus, I'd propose working in a temporary file unless the user specifically provides an output file base (in which case the .R and .md will be appended). This saves some of the utility work in the reprex function, and lets the user choose the filename without being appended with a date.

**Comment by dgrtwo** [comment]
Also, the parameters suggest that the goal is to keep all snippets from your history in a folder. But doesn't this almost contradict the purpose of such snippets? Once they're pasted in GH or SO, they're on the record in the right place. It would be hard for me to imagine looking through all the snippets, rather than going to the GH Issues or SO questions/answers I'd posted.

**Comment by jennybc** [comment]
Thanks for this discussion (and PR)!

Here's where I was going re: storing the code snippet.

Yes agree default behavior should be to leave no trace behind. In which case one might as well work in a temporary directory, with temporary file names. The current code was influenced by some non-default behavior I think I want ... (?)

For the destination folder, consult a package option that, by default, would not exist / be set. But if it was, then the `.R` and possibly `.md` would go there. I would then define this option in my `.Rprofile` and let snippets quietly accumulate there. I actually *do* want to keep a local copy (as I already do for gists, believe it or not) because sometimes you aren't online and, especially for GitHub issues, they aren't always easy to relocate via search. For example, I weigh in on ~50 different private student repos each semester in my course.

As for the file naming, I like the default scheme to involve date because, again, it's easier for finding stuff. They list in chronological order that way, for example. But it's possible I'm just very peculiar.

In conclusion: yes agree reprexes should default to being ephemeral. So agree it's a good idea to make the specific things I want intrude as little as possible on default behavior (and on the code). Am stuck in meetings today, but can look at / merge your PR tomorrow.

**Comment by dgrtwo** [comment]
OK!! Today or tomorrow morning I'll try modifying my PR to allow saving into a separate directory, along the lines .

BTW I've been using this package for Stack Overflow posts and GH issues today and it's *terrific*
